### ISSUE TITLE
suggestion in planner_tools

### ISSUE THREAD

**Issue by DejaYang** [landing-ai/vision-agent#359]
hey, thanks for your project!  I have tried some tasks about image,  all the step1 is like follows :
```
image = load_image('2025-01-0616-02-57---00012.jpg')
claude35_vqa('Can you describe this image? Is there a TV in it and is it turned off?', [image])
suggestion('How can I detect the TV in the image and determine if it is turned off?', [image])
```
the suggestion is wonderful,  would you like to share the code of suggestion in planner_tools?
thanks a lot!

**Comment by giyosphere** [comment]
I can guide you on how to code a suggestion like this in Python using typical libraries like OpenCV and PyTorch or TensorFlow. Here's a general approach

import cv2
import torch
from yolov5 import YOLOv5  # Example of a YOLO-based detector

def load_image(image_path):
    """Loads the image from the specified path."""
    return cv2.imread(image_path)

def detect_tv(image):
    """Detects TVs in the image using a YOLOv5 model."""
    model = YOLOv5('yolov5s.pt')  # Load a small YOLOv5 model
    results = model.predict(image)
    
    # Filter detections for "TV" class
    tv_detections = [det for det in results if det['class'] == 'TV']

    return tv_detections

def is_tv_off(tv_region):
    """Determines if the detected TV is turned off."""
    gray = cv2.cvtColor(tv_region, cv2.COLOR_BGR2GRAY)
    brightness = gray.mean()

    # Heuristic: if brightness is very low, assume TV is off
    return brightness < 30

# Main function
image = load_image('2025-01-0616-02-57---00012.jpg')
tv_detections = detect_tv(image)

for tv in tv_detections:
    x, y, w, h = tv['box']
    tv_region = image[y:y+h, x:x+w]

    if is_tv_off(tv_region):
        print("The TV is detected and is turned off.")
    else:
        print("The TV is detected and is turned on.")

**Comment by dillonalaird** [comment]
> hey, thanks for your project! I have tried some tasks about image, all the step1 is like follows :
> 
> ```
> image = load_image('2025-01-0616-02-57---00012.jpg')
> claude35_vqa('Can you describe this image? Is there a TV in it and is it turned off?', [image])
> suggestion('How can I detect the TV in the image and determine if it is turned off?', [image])
> ```
> 
> the suggestion is wonderful, would you like to share the code of suggestion in planner_tools? thanks a lot!

Hey @DejaYang , thanks for your interest in VisionAgent! Unfortunately we are not releasing the suggestions module, although it is available through the website va.landing.ai. I will be releasing a blog post soon on how it's implemented though! Basically we label a lot of different "design patterns" for vision problems, and then the suggestion module is responsible for retrieving these patterns based off of the image and user query.

**Comment by DejaYang** [comment]
I'm looking forward to your blog.  And thank you very much for your reply and the sharing of the solution.

**Comment by jaytimbadia** [comment]
@dillonalaird is the blog post available now to have a look at the suggestion module patterns?
